src/safeposix/syscalls/net_calls.rs

```diff
@@ -1240,19 +1234,27 @@ impl Cage {
        loop { //we must block manually
            
             // 1. iterate thru readfds
-            let mut res = self.select_readfds(readfds, &mut new_readfds, &mut retval);
-            if res != 0 {return res}
+            if !readfds.is_null() {
```

Comment:
I had asked you to wrap these in an Option as Some() or None() I think that is better practice with Rust here

Reply:
Yeah I agree, that would be good.